src/Microsoft.AspNetCore.Razor.Tools/GenerateCommand.cs

```diff
@@ -180,14 +176,15 @@ private SourceItem[] GetInputItems(string projectDirectory, List<string> sources
             return items;
         }
 
-        private OutputItem[] GenerateCode(RazorTemplateEngine templateEngine, SourceItem[] inputs)
+        private OutputItem[] GenerateCode(RazorProjectEngine projectEngine, SourceItem[] inputs)
         {
             var outputs = new OutputItem[inputs.Length];
-            Parallel.For(0, outputs.Length, new ParallelOptions() { MaxDegreeOfParallelism = 4 }, i =>
+            Parallel.For(0, outputs.Length, new ParallelOptions() { MaxDegreeOfParallelism = Debugger.IsAttached ? 1 : 4 }, i =>
```

Comment:
Added this `Debugger.IsAttached` piece because debugging this code without it is a hugeee pain.